### PR TITLE
Refactor BaseMetrics so that it is not singleton that is explicitly called by the publc/private rest handler; it is now an osgi service

### PR DESCRIPTION
**Pull request by Channyboy**

### CODE REVIEW

**Comment by Channyboy** [comment]
#build

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetrics.java [hunk 1]
```diff
@@ -25,33 +26,45 @@
...
+@Component(service = { BaseMetrics.class }, configurationPolicy = ConfigurationPolicy.IGNORE, immediate = true)
 public class BaseMetrics {
 
-    protected static BaseMetrics baseMetrics = null;
     protected static String BASE = MetricRegistry.Type.BASE.getName();
     public MBeanServer mbs;
     protected static Set<String> gcObjectNames = new HashSet<String>();
 
-    protected static SharedMetricRegistries SHARED_METRIC_REGISTRY;
+    public static SharedMetricRegistries SHARED_METRIC_REGISTRY;
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetrics.java:46]: Why is `SHARED_METRIC_REGISTRY` public?

File: dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetrics.java [hunk 2]
```diff
@@ -25,33 +26,45 @@
...
-        if (baseMetrics == null)
-            baseMetrics = new BaseMetrics();
-        return baseMetrics;
+    @Reference
+    public void setSharedMetricRegistries(SharedMetricRegistries sharedMetricRegistry) {
+        BaseMetrics.SHARED_METRIC_REGISTRY = sharedMetricRegistry;
     }
 
-    protected BaseMetrics() {
+    public BaseMetrics() {
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/BaseMetrics.java:53]: Do you need an empty constructor?

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
The error in the last build was a connection issue win an unrelated test bucket:

The error being:
`java.net.ConnectException: Connection refused (Connection refused)`

**Pull request merged**